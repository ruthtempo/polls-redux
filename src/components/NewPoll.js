import { useState } from "react";
import { connect } from "react-redux";
import { Button } from "../ui/Button";
import { handleSaveNewQuestion } from "../actions/questions";
import { useNavigate } from "react-router-dom";

const NewPoll = ({ authedUser, dispatch }) => {
  const [option1, setOption1] = useState();
  const [option2, setOption2] = useState();
  const navigate = useNavigate();

  const handleChangeO1 = (e) => {
    setOption1(e.target.value);
  };
  const handleChangeO2 = (e) => {
    setOption2(e.target.value);
  };

  const handleClick = () => {
    const newQuestion = {
      optionOneText: option1,
      optionTwoText: option2,
      author: authedUser,
    };
    dispatch(handleSaveNewQuestion(newQuestion));
    navigate("/");
  };

  return (
    <div className="flex justify-center">
      <div className=" flex flex-col w-full 2xl:w-1/4 md:w-2/3 p-4">
        <p className="text-center text-2xl py-3">Create a New Poll</p>
        <input
          placeholder="Option One"
          className="my-2 rounded-md p-2"
          onChange={handleChangeO1}
        />
        <input
          placeholder="Option Two"
          className="my-2 rounded-md p-2"
          onChange={handleChangeO2}
        />
        <Button buttonText={"Save"} onClick={handleClick} />
      </div>
    </div>
  );
};

const mapStateToProps = ({ authedUser }) => ({
  authedUser,
});

export default connect(mapStateToProps)(NewPoll);
