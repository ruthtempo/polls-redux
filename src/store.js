import middleware from "./middleware";
import reducer from "./reducers";
import { createStore } from "redux";

export const store = createStore(reducer, middleware);
