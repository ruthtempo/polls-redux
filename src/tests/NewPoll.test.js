import NewPoll from "../components/NewPoll";
import { store } from "../store";
import { Provider } from "react-redux";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

describe("NewPoll", () => {
  it("will match snapshot", () => {
    const component = render(
      <Provider store={store}>
        <MemoryRouter>
          <NewPoll />
        </MemoryRouter>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });
});
