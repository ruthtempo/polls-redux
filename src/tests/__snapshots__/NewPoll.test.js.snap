// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewPoll will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="flex justify-center"
      >
        <div
          class=" flex flex-col w-full 2xl:w-1/3 md:w-2/3 p-4"
        >
          <p
            class="text-center text-lg py-3 text-gray-500"
          >
            Create a New Poll
          </p>
          <p
            class="text-center text-3xl py-3"
          >
            Would you Rather
          </p>
          <input
            class="my-2 rounded-md p-2"
            placeholder="Option One"
          />
          <input
            class="my-2 rounded-md p-2"
            placeholder="Option Two"
          />
          <button
            class="drop-shadow-md border-solid bg-indigo-500 hover:bg-indigo-700 py-2 px-3 my-2 rounded-full text-white text-center capitalize disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none "
            type="button"
          >
            Save
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex justify-center"
    >
      <div
        class=" flex flex-col w-full 2xl:w-1/3 md:w-2/3 p-4"
      >
        <p
          class="text-center text-lg py-3 text-gray-500"
        >
          Create a New Poll
        </p>
        <p
          class="text-center text-3xl py-3"
        >
          Would you Rather
        </p>
        <input
          class="my-2 rounded-md p-2"
          placeholder="Option One"
        />
        <input
          class="my-2 rounded-md p-2"
          placeholder="Option Two"
        />
        <button
          class="drop-shadow-md border-solid bg-indigo-500 hover:bg-indigo-700 py-2 px-3 my-2 rounded-full text-white text-center capitalize disabled:bg-slate-50 disabled:text-slate-500 disabled:border-slate-200 disabled:shadow-none "
          type="button"
        >
          Save
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
